commcare_connect/templates/opportunity/add_visits_existing_users.html

```diff
@@ -0,0 +1,113 @@
+{% extends "base.html" %}
+{% load static %}
+{% load crispy_forms_tags %}
+
+{% block title %}{{ request.org }} - {{ opportunity.name }}{% endblock %}
+
+{% block content %}
+  <div class="container">
+    <h1 class="display-5">Edit Opportunity</h1>
+    <h4>Adding Visits for Existing Users</h4>
+    <form method="post"
+          x-data="{
+            max_visits: 0,
+            end_date: '',
+            selected_users: [],
+            isSelected (id) { return this.selected_users.find(el => el === id)},
+            calculated_budget() { return this.selected_users.length * this.max_visits * '{{ budget_per_visit }}' }
+          }"
+    >
+      {{ formset.management_form }}
+      {% csrf_token %}
+      <table class="table table-custom border">
+        <thead class="table-secondary">
+        <tr>
+          <th></th>
+          <th>Name</th>
+          <th>Max Visits</th>
+          <th>Used Visits</th>
+          <th>Extend End date</th>
+        </tr>
+        </thead>
+        <tbody>
+        {% for form in formset %}
+          <tr>
+            <td>
+              {% for hidden in form.hidden_fields %}
+                {{ hidden }}
+              {% endfor %}
+              <input type="checkbox" name="{{ form.is_selected.html_name }}"
+                     id="{{ form.is_selected.auto_id }}" value="{{ form.instance.id }}"
+                     x-model="selected_users"/>
+            </td>
+            <td>{{ form.instance.opportunity_access.user.username }}</td>
+            <td>
+              <input type="number" name="{{ form.max_payments.html_name }}" id="{{ form.max_payments.auto_id }}"
+                     class="numberinput form-control {% if form.max_payments.errors %} is-invalid {% endif %}"
+                     :value="isSelected('{{ form.instance.id }}') && max_visits ? max_visits : '{{ form.max_payments.value }}'"
+              />
+              {% for error in form.max_payments.errors %}
+                <div class="invalid-feedback">
+                  <strong>{{ error|escape }}</strong>
+                </div>
+              {% endfor %}
+            </td>
+            <td>{{ form.instance.opportunity_access.visit_count }}</td>
+            <td>
+              <input type="date" name="{{ form.end_date.html_name }}" id="{{ form.end_date.auto_id }}"
+                     class="dateinput form-control{% if form.end_date.errors %} is-invalid{% endif %}"
+                     :value="isSelected('{{ form.instance.id }}') && end_date ? end_date : '{{ form.end_date.value|date:"Y-m-d" }}'"
+              />
+              {{ form.end_date|as_crispy_errors }}
+            </td>
+          </tr>
```

Comment:
How do these elements work (max visits and end date)? From the screenshot it seems like you could adjust them independently. But it also seems like the value is set from the 'global' controls at the bottom of the table.

But at least the JS budget calculation only takes the global max value.

I think we could possibly simplify this a bit if we just use the table to display the data and select the claims to update rather than using them to actually update the claims as well i.e. we could just do a single update query for all selected claims with the updated values.